mne/stats/cluster_level.py

```diff
@@ -1729,3 +1737,310 @@ def summarize_clusters_stc(
     data_summary[:, 0] = np.sum(data_summary, axis=1)
 
     return klass(data_summary, vertices, tmin, tstep, subject)
+
+
+def cluster_test(
+    df: pd.DataFrame,
+    formula: str = None,  # Wilkinson notation formula for design matrix
+    n_permutations: int = 10000,
+    seed: None | int | np.random.RandomState = None,
+    tail: int = 0,  # 0 for two-tailed, 1 for greater, -1 for less
+    n_jobs: int = 1,  # how many cores to use
+    adjacency: tuple = None,
+    max_step: int = 1,  # maximum distance between samples (time points)
+    exclude: list = None,  # exclude no time points or channels
+    step_down_p: int = 0,  # step down in jumps test
+    t_power: int = 1,  # weigh each location by its stats score
+    out_type: str = "indices",
+    check_disjoint: bool = False,
+    buffer_size: int = None,  # block size for chunking the data
+):
+    """
+    Run a cluster permutation test based on formulaic input.
+
+    # currently only supports paired t-test on evokeds or epochs
+
+    Parameters
+    ----------
+    dataframe : pd.DataFrame
+        Dataframe with evoked/epoched data, conditions and subject IDs.
+    formula : str, optional
+        Wilkinson notation formula for design matrix. Default is None.
```

Comment:
this is in fact not optional; you raise an error if formula is `None`. So this should be a required parameter, e.g., no default value.